docs/internal/github_actions.rst

```diff
@@ -0,0 +1,183 @@
+* js0 - enables --no-rapidjson
+
+Feature Mask
+============
+This is a mask in an attempt to keep names shorter.
+* FM-08
+  * --no-built-in-topics
+  * --no-content-subscription
+  * --no-ownership-profile
+  * --no-object-model-profile
+  * --no-persistence-profile
+* FM-1f
+  * --no-built-in-topics
+* FM-2c
+  * --no-content-subscription
+  * --no-object-model-profile
+  * --no-persistence-profile
+* FM-2f
+  * --no-content-subscription
+* FM-37
+  * --no-content-filtered-topics
+
+****************************************
+Introduction to Github Actions Workflows
+****************************************
+
+Our main workflow which dictates our Github Actions run is
+build_and_test.yml. It defines jobs, which are the tasks that
+are run by the CI.
+
+Cacheing
+========
+
+In order to save time on running jobs, a cacheing systems is used.
+This cacheing system creates .tar.xz files which are called artifacts.
+A cache name comparison is run to determine if there is no new commit
+to what is being built, and use an existing artifact if that is the case.
+
+To make CI runs quicker, a seperate build is made for ACE_TAO, rather
+than building it at the same time as OpenDDS. This allows us to cache
+ACE_TAO relatively often due to the stability of the ace6tao2 branch.
+
+Job Types
+=========
+
+There a number of job types that are contained in the built_and_test.yml.
+Where possible, a configuration will contain 3 jobs. The first job that
+is run is the *ACE_TAO_*. This will create an artifact which is used later
+by the OpenDDS build. The second job is *build_*, which uses the previous
+*ACE_TAO_* job to configure and build OpenDDS. This job will then export
+an artifact to be used in the third step. The third step is the *test_*
+job, which runs the appropriate tests for the associated OpenDDS
+configuration.
+
+Certain builds do not follow this 3 step model. Safety builds are done
+in one step due to the cross-compile nature causing problems. Static and Release
+builds have a large footprint and therefore cannot fit the entire test suite onto
+a Github Actions runner.  As a result, they instead build tests during the test
+step, and run different groups of tests in individual jobs. These individual test
+groups are prefixed by *compiler_* which runs the tests/DCPS/Compiler tests, *unit_*
+which runs the unit tests located in tests/DCPS/UnitTest and tests/unit-tests, and
+*messenger_* which runs the tests in tests/DCPS/Messenger and tests/DCPS/C++11/Messenger.
+
+In addition to these builds, there are some builds which will not run the test suite, in
+an effort to shorten the runtime of the continuous integration.  An exception to this is
+that all builds which are not safety, and have ownership profile enabled, will run the
+tests/cmake tests. Test runs which only contain CMake tests are prefixed by *cmake_*.
+
+Test Results
+============
+
+The tests are run using autobuild which will generate an artifact containing the test
```

Comment:
Link to autobuild?

Also, for clarity, something like: Autobuild creates several files which are turned into another GitHub artifact... And this artifact is then analyzed by the "Check Test Results" workflow which will modify the PR's "Checks" results to include detailed summaries of these test runs.